### PR TITLE
Added Ugly Number in Java, Py & Go

### LINKED ISSUE #1075: Ugly number in Python, Java and Go
Please assign it to me.

### PR DESCRIPTION
**Pull request by MastersAbh**
Fixes #1075  

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added code for finding Nth ugly number in Java, Python & Go
- Added code for finding if an input number is an Ugly number in Python and Java

#### Languages Used:
- Java
- Python
- Go

#### Files Added:
- Nth_Ugly_Number.go
- Nth_Ugly_Number.java
- Nth_Ugly_Number.py
- Ugly_Number.java
- Ugly_Number.py

### CODE REVIEW

File: Ugly_Number/Nth_Ugly_Number.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+class Ugly{
```
Comment (singh-shreya6) [inline Ugly_Number/Nth_Ugly_Number.java]: Please provide a brief description of what are ugly numbers across all the files at the start of code

File: Ugly_Number/Nth_Ugly_Number.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+func main() {
+    var number = 100
+    var no = NthUglyNumber(number) 
+    var number_str = strconv.Itoa(number)
+    var no_str = strconv.Itoa(no)
+    fmt.Print("The Ugly no. at position " + number_str + " is " + no_str )
+}
+/*
```
Comment (singh-shreya6) [inline Ugly_Number/Nth_Ugly_Number.go]: Leave a blank line

File: Ugly_Number/Nth_Ugly_Number.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        return i; 
+    } 
+
+    public static void main(String args[]) { 
+        int number = 100;
+        int no = NthUglyNumber(number); 
+        System.out.println("The Ugly no. at position " + number + " is " + no );
+    }
+}
+/*
```
Comment (singh-shreya6) [inline Ugly_Number/Nth_Ugly_Number.java]: Please give a blank line here.

File: Ugly_Number/Nth_Ugly_Number.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            num /= 3  
+        # To check if number is divisible by 2 or not
+        elif (num % 2 == 0) :  
+            num /= 2  
+        else :
+            x = False  
+            break
+    return x
+  
+
```
Comment (singh-shreya6) [inline Ugly_Number/Nth_Ugly_Number.py]: One line gap is sufficient here.

File: Ugly_Number/Nth_Ugly_Number.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    i = 1 
+    count = 1   # ugly number count  
+    # Check for all integers untill ugly count  becomes n 
+    while (n > count) :
+        i = i + 1
+        if ( isUgly(i) ) :
+            count = count + 1 
+    return i
+
+
```
Comment (singh-shreya6) [inline Ugly_Number/Nth_Ugly_Number.py]: One line gap is sufficient

File: Ugly_Number/Ugly_Number.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+                x = true;  
+                break;  
+            }  
+        } 
+        if ( !x ){ 
+            System.out.print("It is an ugly number.");
+        }
+    }
+}
+/*
```
Comment (singh-shreya6) [inline Ugly_Number/Ugly_Number.java]: Give a blank line here.

File: Ugly_Number/Ugly_Number.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import java.util.Scanner;
```
Comment (singh-shreya6) [inline Ugly_Number/Ugly_Number.java]: Add description of problem statement on top

File: Ugly_Number/Nth_Ugly_Number.go [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+            num = num / 2  
+        } else {
+            x = false  
+            break
+        }
+    }
+    return x
+}
+  
+
```
Comment (singh-shreya6) [inline Ugly_Number/Nth_Ugly_Number.go]: One line gap is sufficient

File: Ugly_Number/Nth_Ugly_Number.go [hunk 9]
```diff
@@ -0,0 +1,51 @@
+package main
+
+import(
+    "fmt"
+    "strconv"
+)
+// Function to check if a number is ugly or not
```
Comment (singh-shreya6) [inline Ugly_Number/Nth_Ugly_Number.go]: Give a blank line before comment

File: Ugly_Number/Nth_Ugly_Number.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
+// Ugly numbers are numbers whose only prime factors are 2, 3 or 5.
+class Ugly{
```
Comment (singh-shreya6) [inline Ugly_Number/Nth_Ugly_Number.java]: Give a blank line here

File: Ugly_Number/Nth_Ugly_Number.py [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    while (n > count) :
+        i = i + 1
+        if ( isUgly(i) ) :
+            count = count + 1 
+    return i
+
+number = 100
+no = NthUglyNumber(number) 
+print("The Ugly no. at position " + str(number) + " is " + str(no) , end = '')
+'''
```
Comment (singh-shreya6) [inline Ugly_Number/Nth_Ugly_Number.py]: Blank line here.

File: Ugly_Number/Ugly_Number.py [hunk 12]
```diff
@@ -0,0 +1,21 @@
+# Ugly numbers are numbers whose only prime factors are 2, 3 or 5.
+x = False
```
Comment (singh-shreya6) [inline Ugly_Number/Ugly_Number.py]: Blank line

Comment (MastersAbh) [reply Ugly_Number/Ugly_Number.py]: where ?

Comment (MastersAbh) [reply Ugly_Number/Ugly_Number.py]: do we need to give a blank line after comments too?

Comment (singh-shreya6) [reply Ugly_Number/Ugly_Number.py]: It is the start of code, one line gap after description provides a clear view of where actual code starts

**Comment by MastersAbh** [comment]
Please review @singh-shreya6

**Pull request merged**